src/main/java/io/confluent/connect/elasticsearch/ElasticsearchClient.java

```diff
@@ -715,20 +690,12 @@ private void removeFromInFlightRequests(long executionId) {
   }
 
   /**
-   * Reports a bad record to the DLQ.
+   * Reports a bad record and errors to the DLQ.
    *
    * @param response    the failed response from ES
    * @param executionId the execution id of the request associated with the response
    */
-  private synchronized void reportBadRecord(BulkItemResponse response,
-                                            long executionId) {
-
-    // RCCA-7507 : Don't push to DLQ if we receive Internal version conflict on data streams
-    if (response.getFailureMessage().contains(VERSION_CONFLICT_EXCEPTION)
-            && config.isDataStream()) {
-      log.debug("Skipping DLQ insertion for DataStream type.");
-      return;
-    }
```

Comment:
Did we check why this condition was added and if it is safe to remove this and report error for VERSION_CONFLICT_EXCEPTION as well now?

Reply:
I think this was missed while rebasing. Thanks for pointing this out. I have reverted this change.